### PR TITLE
DM-14255: Add ConvexPolygon.intersects and related methods

### PR DESCRIPTION
**Pull request by r-owen**
Add the following methods to ConvexPolygon:
- contains(Region)
- isDisjointFrom(Region)
- intersects(Region)
- isWithin(Region)

### CODE REVIEW

File: src/ConvexPolygon.cc [hunk 1]
```diff
@@ -313,6 +313,22 @@ bool ConvexPolygon::contains(UnitVector3d const & v) const {
     return detail::contains(_vertices.begin(), _vertices.end(), v);
 }
 
+bool ConvexPolygon::contains(Region const & r) const {
+    return (relate(r) & CONTAINS) != 0;
```
Comment (TallJimbo) [inline src/ConvexPolygon.cc]: I don't think the ` != 0` part of these tests is necessary.  Do you prefer them stylistically? (I don't but, not strongly.)

Comment (smonkewitz) [reply src/ConvexPolygon.cc]: I think it is - `std::bitset` does not have a conversion operator to a `bool`. Another option here is to write `(relate(r) & CONTAINS).any()` instead.

Comment (r-owen) [reply src/ConvexPolygon.cc]: Exactly. I am not fond of `!= 0` but it gets the job done. I am happy to switch to `bitset.any` if you like. I would have preferred to use `bitset.test` but there are no useful constants for that (for good reasons, as @TallJimbo pointed out in email).

Comment (TallJimbo) [reply src/ConvexPolygon.cc]: I suppose I might have a slight preference for `any()`, as it's a helpful reminder to the reader that these are not in fact integers, but it's not a strong preference.

**Pull request merged**